### DISCUSSION TITLE
AIRDROPS package discussions

### DISCUSSION THREAD

**Discussion by Geet-George** [Geet-George/halodrops#34] [General]
## 👋 Welcome!
  We’re using this page as a place to discuss planned developments for the airdrops package. Feel free to:
  * Ask questions you’re wondering about.
  * Share ideas.

**Comment by Geet-George** [comment]
New (potential) name: **AIRDROPS** (**A**pplication for **I**nspecting & **R**estructuring **DROPS**onde data)

**Reply by Geet-George** [reply]
There is a problem with the `airdrops` name. It is [currently taken](https://pypi.org/project/airdrops/) on PyPI. However, there is not much in the repo and hasn't been updated in 1.5 years. As after all our brainstorming, `airdrops` was one name that we all liked and agreed on, I am wondering if I should simply ask [the maintainer](https://pypi.org/user/noirbizarre/) if they could let us use the name. For readthedocs and conda, from a quick search it seems like there are no projects named `airdrops`.

We could think of another name. One I can think of is DRopsonde SOftware for campaigN DatasEts (Dr. SONDE), but it is nowhere near nice as `airdrops` and I would rather not spend more time thinking of another name. So, I am leaning on requesting [the maintainer](https://pypi.org/user/noirbizarre/). Any ideas / opinions?

**Comment by Geet-George** [comment]
## Milestones

- [ ] Up to Level-3 data by 30 June, 2023 -> functions for processing ready in the package. Not necessarily working with inspect & main functions
- [ ] alpha-version of Inspect & main functions working in package by 30 June, 2023
- [ ] All scripts (QC, all levels, plotting) integrated into package by 14 July, 2023

**Reply by Geet-George** [reply]
Thursday, 29th June, 11:00 CEST : Online meeting to check in on progress to the first milestone.

**Comment by Geet-George** [comment]
## Short-term plans

Based on which aspects of the package you are working on, please update here in brief the following:
**1. What's done?**:
**2. What's to be done?**:
**3. When do you plan to finish the pull request for point-2?**:
**4. What is left for it to be working in at least a minimal setting? / What do you need from others?** :

**Reply by Geet-George** [reply]
## Making the package work with an inspect function (#31) - Geet & @hdorff94 

1. What's done?: A working version of the inspect function is ready, and now, `main` is able to detect changes to the default in the config file. (I'll update the PR number here in a while)
2. What's to be done?: Only changes to the default in function arguments can be read, the `MANDATORY` section needs to be incorporated
3. When do you plan to finish the pull request for point-2?: A draft PR should be done by today and it will be readt for review/merging by Thursday.
4. What is left for it to be working in at least a minimal setting? / What do you need from others? : We need other functions to be integrated in to this workflow. In this regard, all functions must ensure that every argument has a default value, and if not, must be covered by the `MANDATORY` section in the config file.

**Reply by Geet-George** [reply]
## Integrating ASPEN-CLI #25 (updating on behalf of @d70-t)

1. **What's done?**: Tobi made a very elegant implementation of using a Docker container to run `aspenqc` (see #25)
2. **What's to be done?**: Decide on how to get ASPEN outpute, i.e. CSV with a simple Python API or NC but deal with tmp files (maybe we can convince ASPEN to provide NC output with `stdout`?)
3. **When do you plan to finish the pull request for point-2?**: Undecided (I'm not sure of Tobi's timeline, and I know at least he is busy this week, so I can help out if he can't manage time next week)
4. **What is left for it to be working in at least a minimal setting? / What do you need from others?** : Geet should figure out I/O aspects for this `aspenqc` integration?

**Reply by hdorff94** [reply]
### **Inspecting routines:**

very busy times and I have to delegate tasks unfortunately as I don't know when I will leave for two months (off topic submissions priotized). I have code snippets of inspecting concerned routines on my branch. Today it is unfortunately unaccessible during Levante Maintenance. @Geet-George Maybe we should discuss if it contains useful add-ons.

**Reply by Geet-George** [reply]
@hdorff94 of course, no worries, I understand. Let's continue the discussion over at #31.

**Reply by ninarobbins** [reply]
## Quicklooks - Nina

**1. What's done:** I currently have [working quicklooks functions](https://github.com/ninarobbins/halodrops/blob/quicklooks/src/halodrops/plotting/quicklooks.py) for the plots in `quickdrops`. I made an [example notebook](https://github.com/ninarobbins/halodrops/blob/quicklooks/docs/source/howto/create_quicklooks.ipynb) and also a [script](https://github.com/ninarobbins/halodrops/tree/quicklooks/src/halodrops/api/plotting.py). 
**2. What's to be done:** adapt the script to be linked to the main script. I am not sure how to adapt it to read function arguments/inputs from the main function. Some additional options such as saving as PDF/setting useful defaults are still not added, but perhaps this is not first priority for an initial working version.
**3. When do you plan to finish the pull request for point-2?:** if I manage to link the scripts to the main function in api, I could maybe be able to finish next week.
**4. What is left for it to be working in at least a minimal setting? / What do you need from others?:** I would need some help/guidance on how to adapt my quicklooks scripts to the structure that was decided on.

**Reply by Geet-George** [reply]
Hi @ninarobbins the functions and notebook look great. Some bits need work, but the bulk is already there 🥳  Do you want to create a draft PR? And I can help review some aspects, including adapting to the main function. 

That said, for the notebook, I think we can keep it in a different PR, as there the objective is to document it in a way that users can quickly get the idea of using it. That might need some explanation and maybe not so much external dependence, such as the `eurec4a_intake`. We'll figure that out slowly.

**Reply by ninarobbins** [reply]
@Geet-George I created a draft pull request, and removed the notebook for now.
